code-challenges.js

```diff
@@ -1,18 +1,56 @@
 // ASSESSMENT 6: JavaScript Coding Practical Questions
 
-
 // --------------------1) Create a function that takes in an array of objects and returns a sentence about each person with their names capitalized.
 
 var people = [
   { name: "ford prefect", occupation: "hitchhiker" },
   { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "radio employee" }
+  { name: "arthur dent", occupation: "radio employee" },
 ]
-// Expected output example: "Ford Prefect is a hitchhiker." "Zaphod Beeblebrox is a president of the galaxy." "Arthus Dent is a radio employee."
-
 
+// function takes an array of objects
+const personJob = (arr) => {
+  // nameArr takes the variable array of people then extracted the names into an array with only the names
+  let nameArr = arr.map((val) => {
+    console.log(val.name)
+    // local var splitting the names up
+    let splitMap = val.name.split(" ")
+    console.log(splitMap)
+    // returning and mapping through the split names, setting the first letter to be uppercase then making a slice at index 1 to not uppercase everything
+    // Then .joining the "strings" again in a str ${interpolation} where val = names and occupation = occupation
+    return (
+      splitMap.map((val) => val[0].toUpperCase() + val.slice(1)).join(" ") +
```

Comment:
I like the simplicity of this. However there is a mix of using string interpolation and string concatenation for logic on the same line/block. Ideally, only one should be used unless it's absolutely unavoidable.